app/models/maths_and_physics/dqt_record.rb

```diff
@@ -0,0 +1,62 @@
+# Used to model a record in DQT for the Maths and Physics policy
+#
+# Should be initialised with data from a row in the report
+# requested from the Database of Qualified Teachers
+#
+# Determines the eligibility of a teacher's qualifications for
+# the Maths and Physics policy.
+#
+#   qts_award_date:     The date the teacher achieved qualified
+#                       teacher status
+#   itt_subject_codes:  The corresponding JAC codes to the subject
+#                       specialism that the teacher competed their
+#                       initial teacher training in
+#   postgraduate_codes: The corresponding JAC codes to the subject(s)
+#                       the teacher completed their degree in
+#
+module MathsAndPhysics
+  class DQTRecord
+    attr_reader :qts_award_date, :itt_subject_codes, :postgraduate_codes
+
+    # Full list of JAC principal subject codes can be found:
+    # https://www.hesa.ac.uk/support/documentation/jacs/jacs3-principal
+    #
+    # Eligible JAC codes for this policy:
+    # https://www.gov.uk/government/publications/additional-payments-for-teaching-eligibility-and-payment-details/claim-a-payment-for-teaching-maths-or-physics-eligibility-and-payment-details#teachers-qualifications
+    ELIGIBLE_JAC_CODES = [
+      "G1", # Mathematics
+      "G2", # Operational research
+      "G3", # Statistics
+      "G9", # Others in mathematical sciences
+      "F3" # Physics
+    ].freeze
+
+    def initialize(data)
+      @qts_award_date = Date.parse(data.fetch("dfeta qtsdate"))
+      @itt_subject_codes = data.slice("ITTSub1Value", "ITTSub2Value", "ITTSub3Value").compact.values
+      @postgraduate_codes = data.slice("HESubject1Value", "HESubject2Value", "HESubject3Value").compact.values
```

Comment:
Yeah I flip-flopped between having them as private methods and in the constructor. I ended up choosing constructor because I thought that surfacing any errors closer to where they are passed in would be better. I'm happy to change it back though

Reply:
Fair enough 👍